include/zephyr/sys/arch_interface.h

```diff
@@ -459,6 +459,17 @@ static inline uint32_t arch_proc_id(void);
  * This will invoke z_sched_ipi() on other CPUs in the system.
  */
 void arch_sched_ipi(void);
+
+/**
+ * @brief Returns the number of CPUs
+ *
+ * For most systems this will be the same as CONFIG_MP_MAX_NUM_CPUS,
+ * however some systems may determine this at runtime instead.
+ *
+ * @return the number of CPUs
+ */
+static inline unsigned int arch_num_cpus(void);
```

Comment:
Should this be `ALWAYS_INLINE`?

Reply:
the other functions in that file don't so I kept with that pattern.

Reply:
No, sorry. Ignore the above.